rpc-client/src/nonblocking/rpc_client.rs

```diff
@@ -2197,12 +2199,20 @@ impl RpcClient {
             current_percent = 100f32 * max as f32 / total_active_stake as f32;
             if current_percent < max_stake_percent {
                 break;
```

Comment:
*style*

If you are going to be fixing this, I suggest writing it completely in the imperative style or completely in the functional style.  A mix of two styles is negatively affecting my ability to read the code, as it does not work fully with either of the existing intuitions that I have.

Imperative approach:
```rust
use itertools::chain;
/* ... */

            let mut max = 0;
            let mut total = 0;
            for account in chain(&vote_accounts.current, &vote_accounts.delinquent) {
                let activated_stake = account.activated_stake;
                max = std::cmp::max(max, activated_stake);
                total += activated_stake;
            }
```

Functional approach:
```rust
use itertools::chain;
/* ... */

            let (max, total) = chain(&vote_accounts.current, &vote_accounts.delinquent)
                .map(|account| account.activated_stake)
                .fold((0, 0), |(max, total), stake| (std::cmp::max(max, stake), total + stake));
```

Reply:
Is it okay if we do this in a follow-up? Hoping to limit to functional change for this commit